url933dd5727b_AppleLiang_Testing_Project_tgz-pJ8-org_joda_time_chrono_gj_MainTestJ8/src/org/joda/time/TestSerialization.java

```diff
@@ -0,0 +1,511 @@
+
+    public void testDuration() throws Exception {
+        Duration test = Duration.millis(12345);
+        loadAndCompare(test, "Duration", false);
+        inlineCompare(test, false);
+    }
+
+    public void testSerializedCopticChronology() throws Exception {
+        CopticChronology test = CopticChronology.getInstance(LONDON);
+        loadAndCompare(test, "CopticChronology", true);
+        inlineCompare(test, true);
+    }
+
+    public void testSerializedISOChronology() throws Exception {
+        ISOChronology test = ISOChronology.getInstance(PARIS);
+        loadAndCompare(test, "ISOChronology", true);
+        inlineCompare(test, true);
+    }
+
+    public void testSerializedGJChronology() throws Exception {
+        GJChronology test = GJChronology.getInstance(TOKYO);
+        loadAndCompare(test, "GJChronology", true);
+        inlineCompare(test, true);
+    }
+
+    public void testSerializedGJChronologyChangedInternals() throws Exception {
+        GJChronology test = GJChronology.getInstance(PARIS, 123L, 2);
+        loadAndCompare(test, "GJChronologyChangedInternals", true);
+        inlineCompare(test, true);
+    }
+
+    public void testSerializedGregorianChronology() throws Exception {
+        GregorianChronology test = GregorianChronology.getInstance(PARIS);
+        loadAndCompare(test, "GregorianChronology", true);
+        inlineCompare(test, true);
+    }
+
+    public void testSerializedJulianChronology() throws Exception {
+        JulianChronology test = JulianChronology.getInstance(PARIS);
+        loadAndCompare(test, "JulianChronology", true);
+        inlineCompare(test, true);
+    }
+
+    public void testSerializedBuddhistChronology() throws Exception {
+        BuddhistChronology test = BuddhistChronology.getInstance(PARIS);
+        loadAndCompare(test, "BuddhistChronology", true);
+        inlineCompare(test, true);
+    }
+
+    public void testSerializedPeriodType() throws Exception {
+        PeriodType test = PeriodType.dayTime();
+        loadAndCompare(test, "PeriodType", false);
+        inlineCompare(test, false);
+    }
+
+    public void testSerializedDateTimeFieldType() throws Exception {
+        DateTimeFieldType test = DateTimeFieldType.clockhourOfDay();
+        loadAndCompare(test, "DateTimeFieldType", true);
+        inlineCompare(test, true);
+    }
+
+    public void testSerializedUnsupportedDateTimeField() throws Exception {
+        UnsupportedDateTimeField test = UnsupportedDateTimeField.getInstance(
+                DateTimeFieldType.year(),
+                UnsupportedDurationField.getInstance(DurationFieldType.years()));
+        loadAndCompare(test, "UnsupportedDateTimeField", true);
+        inlineCompare(test, true);
+    }
+
+    private void loadAndCompare(Serializable test, String filename, boolean same) throws Exception {
+        FileInputStream fis = new FileInputStream("src/resources/" + filename + ".dat");
+        ObjectInputStream ois = new ObjectInputStream(fis);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *ois*. The resource is referenced at lines 294, 295. The resource is closed at line 296. There are other execution paths that do not contain closure statements, for example, when ObjectInputStream.readObject throws an exception. Ensure *ois* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).